src/endpoint/s3/s3_utils.js

```diff
@@ -131,9 +133,27 @@ function get_request_xattr(req) {
 }
 
 function set_response_xattr(res, xattr) {
-    _.each(xattr, (val, key) => {
-        res.setHeader(X_AMZ_META + key, val);
-    });
+    let size_for_md_left = config.S3_MD_SIZE_LIMIT;
+    const keys = Object.keys(xattr);
+    // eslint-disable-next-line no-prototype-builtins
+    if (xattr.hasOwnProperty(XATTR_SORT_SYMBOL)) {
```

Comment:
I think you can just use brackets to access it, and then there's no need to disable eslint.
In any case we should definitely add a comment on this code block to explain *why* we introduced this symbol and condition.
```suggestion
    // Using a special symbol on the xattr object to specify when sorting of keys is needed
    // which will make sure the returned headers are consistently being truncated.
    // The symbol is set by namespace_fs and could be used in other namespaces where
    // xattrs are more likely to exceed the size limit and therefore sorting is desired,
    // but for most namespaces where the size is also imposed on PUT, we can avoid it.
    if (xattr[XATTR_SORT_SYMBOL]) {
```